llvm/include/llvm/Analysis/ValueTracking.h

```diff
@@ -39,7 +40,6 @@ struct KnownBits;
 class Loop;
 class LoopInfo;
 class MDNode;
-struct SimplifyQuery;
```

Comment:
Can the forward declaration be retained instead of including the header here?

Reply:
This is possible, but my thinking here was that (at least after more APIs have been migrated) actually using any ValueTracking functions will need SimplifyQuery.h anyway, so it would be more convenient if ValueTracking.h directly pulls it in, instead of including it in each user.

Including it also allows using the implicit DataLayout -> SimplifyQuery conversion, though I'm not sure how much people use that.

Reply:
Any thoughts on this?

Reply:
I think in general we often try to limit includes in headers if forward declarations are sufficient. I don't have any strong feelings about this case, if it is overall beneficial to include it in the header that would be fine.

Reply:
Okay, if nobody feels strongly about this, I'd prefer to keep the current version.